src/ari_sxn_simbeamline/xrt_sim/ari_sim.py

```diff
@@ -0,0 +1,674 @@
+        had been changed and then updates the outcome of Beam Object accordingly.
+
+
+        Parameters
+        ----------
+        update: a dict wrapping up the PV names and values,
+                update = {'ARI_M2:Ry_fine':0.2}.
+
+        updated: a boolean, i.e., False (by default) or True.
+        The Ture means the outcome of Beam Object needs to be updated
+        as the beamline aperature has been modified.
+
+        Returns
+        -------
+        updated : Boolean
+            Potentially modified the input parameter updated if the update
+            indicates a re-activation required.
+
+        '''
+
+        if update != None:
+            for pv_name, pv_val in update.items():
+                if pv_name.split(':')[-1] in ['x', 'y', 'z']:
+                    center_list = getattr(self, 'center').copy() # self.pv2xrt[pv_name.split(':')[0]][pv_name]
+
+                    # Find out the position of changed element based on the
+                    # coordination transformation between NSLS2 and XRT
+                    unit_vector_trans_NSLS2 = np.array([0, 0, 0])
+                    for i, p in enumerate(['x','y','z']):
+                        if pv_name.split(':')[-1] == p:
+                            unit_vector_trans_NSLS2[i] = 1.0
+                    unit_vector_trans_XRT = np.dot(self.coordinate_NSLS2XRT['upward'],
+                                                   unit_vector_trans_NSLS2)
+
+                    center_list[int(np.where(unit_vector_trans_XRT!=0)[0][0])] = (
+                            pv_val * unit_vector_trans_XRT[int(np.where(unit_vector_trans_XRT!=0)[0][0])])
+                    if getattr(self, 'center') != center_list:
+                        setattr(self, 'center', center_list) # self.pv2xrt[pv_name.split(':')[0]][pv_name]
+                        updated = True
+                else:
+                    if getattr(self, self.pv2xrt[pv_name.split(':')[0]][pv_name]) != pv_val:
+                        setattr(self, self.pv2xrt[pv_name.split(':')[0]][pv_name], pv_val)
+                        updated = True
+
+        if updated:
+            self.beamIn = getattr(self._upstream, 'beamOut')
+            self.beamOut = self.expose(self.beamIn)
+
+        return updated
+
+
+
+
+class AriModel:
+    """
+    The ARI beamline simulation based on XRT.
+
+    This class simulates the beam propagation along the ARI beamline and gives
+    the beam properties from each beamline component.
+
+    Parameters
+    ----------
+    photon_energy : argument,specify an initial photon energy for the simulation
+
+    polarization: argument, specify the polarization for simulation.
+
+    Attributes
+    ----------
+    *attrs : many
+
+    Methods
+    -------
+    *methods : many
+
+    """
+
+    def __init__(self, update_comp=None):
+        self.update_comp = update_comp
+
+    ###############################################
```

Comment:
I prefer to not have these comment lines can you remove them here and the two below?